src/lib.rs

```diff
@@ -1,3 +1,75 @@
+//! Landlock is a security feature available since Linux 5.13.
+//! The goal is to enable to restrict ambient rights
+//! (e.g., global filesystem access)
+//! for a set of processes by creating safe security sandboxes as new security layers
+//! in addition to the existing system-wide access-controls.
+//! This kind of sandbox is expected to help mitigate the security impact of bugs,
+//! unexpected or malicious behaviors in applications.
+//! Landlock empowers any process, including unprivileged ones, to securely restrict themselves.
+//! More information about Landlock can be found in the [official website](https://landlock.io).
+//!
+//! This crate provides a safe abstraction for the Landlock system calls, along with some helpers.
+//!
+//! # Use cases
+//!
+//! This crate is especially useful for build-in application sandboxing:
+//! * Parsers hardening (e.g., archive tools, file format conversion, renderers).
+//! * (Part of) applications with limited file renaming or linking needs
+//!   (e.g., some system or network services).
+//! * Monolith applications dealing with different levels of confidentiality
+//!   (e.g., web browser, email server).
```

Comment:
+1, good examples; (web browsers already use seccomp, and I believe qmail has such a compartmentalized architecture); Some terminlogogy nitpicking: I'm not sure if these still qualify as monoliths though... it's a loaded term, and actually to use process-level sandboxing, the design often needs to be split into multiple processes with different permissions (like qmail)

Reply:
Yep, I'll remove "monolith" which is useless anyway.